mds/data_source_instance_types.go

```diff
@@ -125,25 +127,57 @@ func (d *instanceTypesDataSource) Schema(_ context.Context, _ datasource.SchemaR
 	}
 }
 
-// Read refreshes the Terraform state with the latest data.
+//Read refreshes the Terraform state with the latest data.
+
 func (d *instanceTypesDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
-	var state instanceTypesDataSourceModel
+	var state *instanceTypesDataSourceModel
 
 	// Read Terraform configuration data into the model
 	resp.Diagnostics.Append(req.Config.Get(ctx, &state)...)
-	serviceInstanceTypes, err := d.client.Controller.GetServiceInstanceTypes(&controller.MdsInstanceTypesQuery{
-		ServiceType: service_type.RABBITMQ,
-	})
+
+	if err := service_type.ValidateRoleType(state.ServiceType.ValueString()); err != nil {
+		resp.Diagnostics.AddError(
+			"invalid type",
+			err.Error())
+		return
+	}
+	query := &controller.MdsInstanceTypesQuery{
+		ServiceType: state.ServiceType.ValueString(),
+	}
+	serviceInstanceTypes, err := d.client.Controller.GetServiceInstanceTypes(query)
+
 	if err != nil {
 		resp.Diagnostics.AddError(
-			"Unable to Read MDS InstanceTypes",
+			"Unable to Read TDH Service Types",
 			err.Error(),
 		)
 		return
 	}
 
 	// Map response body to model
 	for _, instanceType := range serviceInstanceTypes.InstanceTypes {
+
+		var nodes int64
+		var maxconnections int64
+
+		parsedNodes, err := strconv.ParseInt(instanceType.Metadata.Nodes, 10, 64)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Failed to parse Nodes value",
+				err.Error(),
+			)
+		}
+		nodes = parsedNodes
+
+		parsedMaxconnections, err := strconv.ParseInt(instanceType.Metadata.MaxConnections, 10, 64)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Failed to parse MaxConnections value",
+				err.Error(),
+			)
+		}
+		maxconnections = parsedMaxconnections
```

Comment:
will this not break after block if there is error, and parsedMaxconnections is nil?

Reply:
Addressed